### ISSUE TITLE
Cleanup `No SOURCES given to Zephyr library` warnings 

### ISSUE THREAD

**Issue by tejlmand** [zephyrproject-rtos/zephyr#38403]
The https://github.com/zephyrproject-rtos/zephyr/pull/37512 cleanup PR created several new Zephyr libraries instead of adding sources to libzephyr.a.

Unfortunately, this also results in some libraries not having sources added to them in certain Kconfig combinations, resulting in the following warning being printed:
```
  No SOURCES given to Zephyr library: drivers__bluetooth

  Excluding target from build.
```

**Expected behavior**
No warning being printed

**Impact**
Annoyance, and bad user experience.

**Comment by Thalley** [comment]
Related to https://github.com/zephyrproject-rtos/zephyr/issues/37770

**Comment by galak** [comment]
It seems like there are two possible approaches from a Kconfig point of view.

1. introduce some form of `<FOO>_HAS_DRIVER` that gets selected by all the driver Kconfig symbols and is used to build the dir/lib.  (See `SERIAL_HAS_DRIVER` for example)
2. rework Kconfig's so things are menuconfig for each driver class, etc).  Has ripple in how various boards/soc's have been selecting drivers

**Comment by tejlmand** [comment]
>     1. introduce some form of `<FOO>_HAS_DRIVER` that gets selected by all the driver Kconfig symbols and is used to build the dir/lib.  (See `SERIAL_HAS_DRIVER` for example)

I have been considering such approach, but I feel it's like asking Kconfig to know about Zephyr CMake internals, as it starts top have symbols intended to help CMake.
Why should kconfig select a promptless symbol just to make things cleaner in CMake.

Of course, if such symbol is available for internal Kconfig purposes, like the `SERIAL_HAS_DRIVER`, then CMake can of course use such knowledge.

That being said, it could end-up being the cleanest solution.

> 2. rework Kconfig's so things are menuconfig for each driver class, etc). Has ripple in how various boards/soc's have been selecting drivers

yes, this is a broader restructure, but could end-up with a more uniform way of selecting drivers in Kconfig in general.

**Comment by tejlmand** [comment]
> > 1. rework Kconfig's so things are menuconfig for each driver class, etc). Has ripple in how various boards/soc's have been selecting drivers
> 
> yes, this is a broader restructure, but could end-up with a more uniform way of selecting drivers in Kconfig in general.

Actually, majority of drivers are already using a `menuconfig`, only a fex exceptions like Bluetooth, interrupt, and Timer drivers.
![image](https://user-images.githubusercontent.com/9802583/132837051-7b6ef9e1-1f18-40c6-91cd-61ac8101e041.png)

So maybe we should use this opportunity to align those few to the rest.

**Comment by tejlmand** [comment]
still a pending PR to be fully fixed: #38908
